### PR TITLE
Push events to (old) talks.ox

### PR DESCRIPTION
**Pull request by martinfilliau**
* [x] create a talk (deal with fields not mandatory)
* [x] update a talk (rather than creating a new one)
* [x] delete a talk (pending decisions on #59 and #65)
* [x] ability to handle series

### CODE REVIEW

**Comment by martinfilliau** [comment]
@ahaith I'm done with this, would be great if you could review it and highlight the needs for documentation..
@envycontent http://talks-ox-dev.nsms.ox.ac.uk/talk/index/33 is an example of talk published using the old talks api fyi

File: talks/old_talks/tasks.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+def update_old_talks(event):
+    if _old_talks_configured():
+        if event.group:
+            old_series, new_group = OldSeries.objects.get_or_create(group=event.group)
+            if new_group:
+                group = group_to_old_series(event.group)
```
Comment (ahaith) [inline talks/old_talks/tasks.py]: This looks like the only place we back-convert to old series, which means that: 
1) A series won't show up on old talks until it contains an event.
2) If its title changes, that won't propagate to old talks.

**Comment by envycontent** [comment]
Great, I'll start trying this out, along with everything else! It will
be worth documenting in the User Guide as well I think, just to explain
the process to people.

Anne

**Comment by martinfilliau** [comment]
@envycontent from what I can see in https://github.com/ox-it/talks-oxford-talks-ruby/blob/master/app/controllers/list_controller.rb it sounds like we cannot update a list -- in the eventuality of a series being updated (e.g. changed title), do you have more info? At the moment, series are only created in the old system when the first event attached to the group is saved, series are never updated.

**Comment by envycontent** [comment]
On 09/01/2015 17:49, Martin Filliau wrote:
> @envycontent <https://github.com/envycontent> from what I can see in
> https://github.com/ox-it/talks-oxford-talks-ruby/blob/master/app/controllers/list_controller.rb
> it sounds like we cannot update a list -- in the eventuality of a series
> being updated (e.g. changed title), do you have more info? At the
> moment, series are only created in the old system when the first event
> attached to the group is saved, series are never updated.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ox-it/talks.ox/pull/84#issuecomment-69370638>.
> 

Hum, interesting. Yes, the old talks aggregator does an annoying thing
of making a brand new list if the title of the list changes (but this is
because there is no other identifier for the incoming lists). So I
imagine this was something that Richard didn't need to tackle rather
than something that was impossible.

Looking at the talks controller there's one update method which seems to
be used for both the api and the user interface. Is it the case the list
update method here just won't work via the api?:

https://github.com/ox-it/talks-oxford-talks-ruby/blob/master/app/controllers/list_controller.rb#L44

If it doesn't then I could probably hack in another api_update method
that would remove the 'flash' and 'redirect_to' and provide the
responses you need.

Alternatively we could tell people that if they change the title of
their list on the new talks they need to go an change it on the old
talks as well.

Anne

**Comment by martinfilliau** [comment]
@envycontent you're right it does work with `update` as long as you ignore the response (which is fine, just checking if it is HTTP 200 OK) so it works now, have a look to http://talks-ox-dev.nsms.ox.ac.uk/show/index/39 I've tested adding multiple talks to the same series and updating the title of the series, everything seems fine.
@ahaith I believe all your comments have been addressed, would be great if you could have another look.

File: talks/old_talks/tasks.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                old_series.save()
+            else:
+                raise Exception(response.status_code)
+        elif force_update:
+            group_xml = group_to_old_series(group)
+            group_url = "{server}/list/update/{id}".format(server=settings.OLD_TALKS_SERVER, id=old_series.old_series_id)
+            response = requests.post(group_url, group_xml, auth=(settings.OLD_TALKS_USER, settings.OLD_TALKS_PASSWORD),
+                                     allow_redirects=True, stream=False, headers={"Accept": "application/xml"})
+            if not response.status_code == 200:
+                # response is a redirection to an edit page so ignore the content...
```
Comment (ahaith) [inline talks/old_talks/tasks.py]: If changing a list title creates a brand new list, don't we need to get the new list ID somehow?

Comment (martinfilliau) [inline talks/old_talks/tasks.py]: We are using the `update/{id}` method here so this won't create a list anyway because the ID would not have been found.

**Pull request merged**